modules/homa/src/lib.rs

```diff
@@ -627,10 +635,20 @@ pub mod module {
 		) -> sp_std::result::Result<R, E> {
 			StakingLedgers::<T>::try_mutate_exists(sub_account_index, |maybe_ledger| {
 				let mut ledger = maybe_ledger.take().unwrap_or_default();
+				let bonded_amount = ledger.bonded;
+
 				f(&mut ledger).map(move |result| {
 					*maybe_ledger = if ledger == Default::default() {
+						TotalStakingBonded::<T>::mutate(|staking_balance| {
+							*staking_balance = staking_balance.saturating_sub(bonded_amount)
+						});
 						None
 					} else {
+						TotalStakingBonded::<T>::mutate(|staking_balance| {
+							*staking_balance = staking_balance
+								.saturating_add(ledger.bonded)
+								.saturating_sub(bonded_amount)
```

Comment:
```suggestion
```

Reply:
The code should be deleted here. What did I miss?

Reply:
It is `TotalStakingBonded` + updated ledger bonded - old ledger bonded.  

Naming it `old_bonded_amount` may be more clear